src/main/java/uk/gov/hmcts/divorce/common/service/task/SetStateAfterSubmission.java

```diff
@@ -33,20 +33,14 @@ public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> cas
             || isSoleApplication && application.isPersonalServiceMethod();
         final boolean isApplicant2AwaitingDocuments = application.hasAwaitingApplicant2Documents();
 
-        if (isSoleApplication && isHWFApplicant1) {
-            if (isApplicant1AwaitingDocuments) {
-                caseDetails.setState(AwaitingDocuments);
-            } else {
-                caseDetails.setState(AwaitingHWFDecision);
-            }
-        } else if (!isSoleApplication && isHWFApplicant1 && isHWFApplicant2) {
-            if (isApplicant1AwaitingDocuments  || isApplicant2AwaitingDocuments) {
-                caseDetails.setState(AwaitingDocuments);
-            } else {
-                caseDetails.setState(AwaitingHWFDecision);
-            }
-        } else if (isApplicant1AwaitingDocuments || !isSoleApplication && isApplicant2AwaitingDocuments) {
+        boolean applicantIsAwaitingDocuments = (isApplicant1AwaitingDocuments && !isHWFApplicant1) ||
+                (!isSoleApplication && isApplicant2AwaitingDocuments && !isHWFApplicant2);
+        boolean applicantNeedsHelpWithFees = isHWFApplicant1 || (!isSoleApplication && isHWFApplicant2);
+
+        if (applicantIsAwaitingDocuments) {
             caseDetails.setState(AwaitingDocuments);
+        } else if (applicantNeedsHelpWithFees) {
```

Comment:
likely simpler to bring the if on applicantNeedsHelpWithFees first?

something like:
`boolean applicantNeedsHelpWithFees = isHWFApplicant1 || (!isSoleApplication && isHWFApplicant2);
boolean applicantIsAwaitingDocuments = isApplicant1AwaitingDocuments || (!isSoleApplication && isApplicant2AwaitingDocuments);

if (applicantNeedsHelpWithFees) {
    caseDetails.setState(AwaitingHWFDecision);
} else if (applicantIsAwaitingDocuments) {
    caseDetails.setState(AwaitingDocuments);
}`